### PR TITLE
Additional options for no-CAPTCHA variant

### PR DESCRIPTION
**Pull request by vaxul**
Hi @mblaschke 

Please have a look and merge if it's okay to you.

Regards

vaxul

### CODE REVIEW

File: class.tx_jmrecaptcha.php [hunk 1]
```diff
@@ -79,12 +99,37 @@ protected function initConfiguration() {
...
 	protected function renderNoCaptcha() {
-		$content = '<script type="text/javascript" src="' . htmlspecialchars($this->conf['server']) . '.js?hl=' . htmlspecialchars($this->conf['lang']) . '"></script>';
-		$content .= '<div class="g-recaptcha" data-sitekey="' . htmlspecialchars($this->conf['public_key']) . '" data-theme="' . htmlspecialchars($this->conf['theme']) . '"></div>';
+		$language = $this->getLanguageCode(null);
+		$languageParameter = '';
+		if (!empty($language)) {
+			$languageParameter = 'hl=' . $language;
+		}
+
+		$content = '<script type="text/javascript" src="' . htmlspecialchars($this->conf['server']) . '.js?' . $languageParameter . '"></script>';
```
Comment (mblaschke) [inline class.tx_jmrecaptcha.php]: why not `htmlspecialchars($this->conf['server'] . '.js?' . $languageParameter)`?

File: class.tx_jmrecaptcha.php [hunk 2]
```diff
@@ -169,6 +205,29 @@ protected static function jsQuote($value) {
...
+	 */
+	protected function getLanguageCode($fallback = 'en') {
+		$languageCode = '';
+
+		if (!empty($fallback)) {
+			$languageCode = $fallback;
+		}
+		if (!empty($this->conf['lang'])) {
+			// language from plugin configuration
+			$languageCode = htmlspecialchars($this->conf['lang']);
```
Comment (mblaschke) [inline class.tx_jmrecaptcha.php]: remove htmlspecialchars here, should be added when including in html :)

**Comment by vaxul** [comment]
Good morning, I'm with you and changed the commented lines.

**Comment by mblaschke** [comment]
Seems to be ok but currently i can't test it. Should we also push this to TER?

**Comment by vaxul** [comment]
In an ideal world, at least one other person would test this. Hard to consider. As far as I can tell, I use this code for production.

Side question: How do you setup custom forms with Google reCAPTCHA within TYPO3?

**Pull request merged**